vendor/bat-native-ledger/src/bat/ledger/internal/gemini/gemini_transfer.h

```diff
@@ -50,7 +50,7 @@ class GeminiTransfer {
                                    const int attempts,
                                    client::TransactionCallback callback);
 
-  base::OneShotTimer retry_timer_;
+  std::map<std::string, std::unique_ptr<base::OneShotTimer>> retry_timer_;
```

Comment:
Any particular reason to use `std::unique_ptr<>`?

Reply:
It's easy to manage scope with a unique ptr, I found this convention used in ads_service_impl as well - https://github.com/brave/brave-core/blob/master/components/brave_ads/browser/ads_service_impl.cc#L1926

Let me know if you have any concerns.

Reply:
Yeah, I'm not quite sure why it's used there either. 🙂
In a `std::map<std::string, base::OneShotTimer>`, the lifetime of  `base::OneShotTimer`s is already managed by the red-black tree implementation of the `std::map<>`. Storing `std::unique_ptr<base::OneShotTimer>`s instead results in unnecessary heap allocations/deallocations.

Reply:
ah! this is helpful, I'll update. Thanks @szilardszaloki